### ISSUE TITLE
Meta build resolves to unused Scala 2.10.2

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#1527]
## steps
1. Use sbt 0.13.5 with the following plugins:

`project/plugins.sbt`:

``` scala
addSbtPlugin("me.lessis" % "bintray-sbt" % "0.1.2")

addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.1.6")

ivyPaths :=  new IvyPaths((baseDirectory in ThisBuild).value, Some((baseDirectory in ThisBuild).value / "ivy-cache"))
```

`build.sbt`:

``` scala
ivyPaths :=  new IvyPaths((baseDirectory in ThisBuild).value / "project", Some((baseDirectory in ThisBuild).value / "project" / "ivy-cache"))
```

Run:

```
$ sbt exit
```
## problem

The meta build resolves to Scala 2.10.2 and downloads it.
## expectation

The meta build uses Scala 2.10.4 (or which ever Scala version the sbt 0.13.x uses)

**Comment by eed3si9n** [comment]
The workaround I posted in #1494:

``` scala
libraryDependencies += "org.scala-lang" % "scala-compiler" % "2.10.4"
```

This seems to be working for me on sbt 0.13.5:

```
Before: sbt exit  18.17s user 1.61s system 13% cpu 2:31.44 total
After:  sbt exit  17.52s user 1.37s system 17% cpu 1:47.75 total
```

```
$ ls project/ivy-cache/cache/org.scala-lang/**/jars/**
project/ivy-cache/cache/org.scala-lang/jline/jars/jline-2.10.4.jar                    project/ivy-cache/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.10.4.jar
project/ivy-cache/cache/org.scala-lang/scala-actors/jars/scala-actors-2.10.2.jar      project/ivy-cache/cache/org.scala-lang/scala-swing/jars/scala-swing-2.10.2.jar
project/ivy-cache/cache/org.scala-lang/scala-compiler/jars/scala-compiler-2.10.4.jar  project/ivy-cache/cache/org.scala-lang/scalap/jars/scalap-2.10.0.jar
project/ivy-cache/cache/org.scala-lang/scala-library/jars/scala-library-2.10.4.jar
```

/cc @paulp

**Comment by paulp** [comment]
I might have been misled by the ongoing appearance of these artifacts, for which there are 2.10.4 versions. I realize that depending on the 2.10.4 library will not "back-propagate" the version to these, but in the haze of ivy output I must have seen the compiler jar everywhere.

```
[info] downloading http://repo1.maven.org/maven2/org/scala-lang/scalap/2.10.0/scalap-2.10.0.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scalap;2.10.0!scalap.jar (11835ms)
[info] downloading http://repo1.maven.org/maven2/org/scala-lang/scala-actors/2.10.2/scala-actors-2.10.2.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scala-actors;2.10.2!scala-actors.jar (6130ms)
[info] downloading http://repo1.maven.org/maven2/org/scala-lang/scala-swing/2.10.2/scala-swing-2.10.2.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scala-swing;2.10.2!scala-swing.jar (9806ms)
```

**Comment by paulp** [comment]
I also might have scoped it to "scala-tool", foolishly it seems.

**Comment by eed3si9n** [comment]
Closing as duplicate of #1439.
